src/System.Management.Automation/engine/remoting/common/RemoteSessionNamedPipe.cs

```diff
@@ -92,11 +99,32 @@ internal static class NamedPipeUtils
                 appDomainName = DefaultAppDomainName;
             }
 
-            return NamedPipeNamePrefix +
-                    proc.StartTime.ToFileTime().ToString(CultureInfo.InvariantCulture) + "." +
-                    proc.Id.ToString(CultureInfo.InvariantCulture) + "." +
-                    CleanAppDomainNameForPipeName(appDomainName) + "." +
-                    proc.ProcessName;
+            System.Text.StringBuilder pipeNameBuilder = new System.Text.StringBuilder(MaxNamedPipeNameSize);
+            pipeNameBuilder.Append(NamedPipeNamePrefix)
+#if UNIX
+                // The starttime is there to prevent another process easily guessing the pipe name
+                // and squatting on it.
+                // There is a limit of 104 characters in total including the temp path to the named pipe file
+                // on non-Windows systems, so we'll convert the starttime to hex and just take the first 8 characters.
+                .Append(proc.StartTime.ToFileTime().ToString("X8").Substring(1,8))
+#else
+                .Append(proc.StartTime.ToFileTime().ToString(CultureInfo.InvariantCulture))
+#endif
+                .Append('.')
+                .Append(proc.Id.ToString(CultureInfo.InvariantCulture))
+                .Append('.')
+                .Append(CleanAppDomainNameForPipeName(appDomainName))
+                .Append('.')
+                .Append(proc.ProcessName);
+#if UNIX
+            int charsToTrim = pipeNameBuilder.Length - MaxNamedPipeNameSize;
+            if (charsToTrim > 0)
+            {
+                pipeNameBuilder.Remove(MaxNamedPipeNameSize + 1, charsToTrim);
+            }
```

Comment:
This operation might cause the pipe file to not be found by the pipe discovery code above. I guess that would be rare and we can deal with it such an issue comes up?

Reply:
I agree.  We should ensure the pipe file pathname is parse-able or it is an error (to log?).

Reply:
@daxian-dbw I considered that, but it does seem rare unless we start having hosts with long filenames like `powershell_ise`...

Due to the length restriction on the pipe path and the current design to embed metadata in the filename, I don't see a clear solution to this other than creating a different file that maps guid named pipe files.  But I would defer that until there's an actual issue.

Reply:
I'm fine that we leave it as is until a real issue comes up, which should be rare.